Gordon360/Services/MembershipService.cs

```diff
@@ -88,167 +92,108 @@ public MEMBERSHIP GetSpecificMembership(int membershipID)
             if (account == null)
             {
                 throw new ResourceNotFoundException() { ExceptionMessage = "The Account was not found." };
             }
 
-            var memberships = await _context.Procedures.MEMBERSHIPS_PER_STUDENT_IDAsync(int.Parse(account.gordon_id));
-
-            return memberships.OrderByDescending(x => x.StartDate).Select(m => new MembershipViewModel
-            {
-                MembershipID = m.MembershipID,
-                ActivityCode = m.ActivityCode.Trim(),
-                ActivityDescription = m.ActivityDescription.Trim(),
-                ActivityImagePath = m.ActivityImagePath,
-                SessionCode = m.SessionCode.Trim(),
-                SessionDescription = m.SessionDescription.Trim(),
-                IDNumber = m.IDNumber,
-                FirstName = m.FirstName.Trim(),
-                LastName = m.LastName.Trim(),
-                Participation = m.Participation.Trim(),
-                ParticipationDescription = m.ParticipationDescription.Trim(),
-                StartDate = m.StartDate,
-                EndDate = m.EndDate,
-                Description = m.Description,
-                ActivityType = m.ActivityType.Trim(),
-                ActivityTypeDescription = m.ActivityTypeDescription.Trim(),
-                GroupAdmin = m.GroupAdmin,
-                Privacy = m.Privacy,
-            });
+            return _context.MembershipView.Where(m => m.Username == account.AD_Username).OrderByDescending(x => x.StartDate);
         }
 
         /// <summary>
         /// Fetches the number of followers associated with the activity whose code is specified by the parameter.
         /// </summary>
         /// <param name="activityCode">The activity code.</param>
         /// <returns>int.</returns>
-        public async Task<int> GetActivityFollowersCountAsync(string activityCode)
+        public int GetActivityFollowersCount(string activityCode)
         {
-            var memberships = await GetMembershipsForActivityAsync(activityCode);
-            return memberships.Where(x => x.Participation == "GUEST").Count();
+            return _context.MembershipView.Where(m => m.ActivityCode == activityCode && m.Participation == "GUEST").Count();
         }
 
         /// <summary>
         /// Fetches the number of memberships associated with the activity whose code is specified by the parameter.
         /// </summary>
         /// <param name="activityCode">The activity code.</param>
         /// <returns>int.</returns>
-        public async Task<int> GetActivityMembersCountAsync(string activityCode)
+        public int GetActivityMembersCount(string activityCode)
         {
-            var memberships = await GetMembershipsForActivityAsync(activityCode);
-            return memberships.Where(x => x.Participation != "GUEST").Count();
+            return _context.MembershipView.Where(m => m.ActivityCode == activityCode && m.Participation != "GUEST").Count();
```

Comment:
Same as above, I think this should be for the current session, not all sessions, right? Can it be combined with the `ForSession` one below?